zpay32/invoice.go

```diff
@@ -606,6 +607,11 @@ func validateInvoice(invoice *Invoice) error {
 		return fmt.Errorf("missing feature vector")
 	}
 
+	// Validate any transitive feature depdepencies.
+	if err := feature.ValidateDeps(invoice.Features); err != nil {
+		return err
+	}
```

Comment:
Do we want to be so strict when parsing payment requests? Probably better to be strict, to force implementations to implement it right.

Reply:
yes i think it's also good to catch this before doing any sort of path finding